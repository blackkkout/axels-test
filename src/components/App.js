import { createTheme, ThemeProvider, CssBaseline } from '@mui/material';

import { Root } from '../styled/App';
import { palette } from '../styled/config';
import { Order } from './Order';

const theme = createTheme({
  palette,
});

export const App = () => (
  <ThemeProvider theme={theme}>
    <CssBaseline />
    <Root>
      <Order />
    </Root>
  </ThemeProvider>
);
