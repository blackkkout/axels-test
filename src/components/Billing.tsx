import { useState } from 'react';
import {
  Button,
  FormControl,
  FormLabel,
  Stack,
  Typography,
  TextField,
  Autocomplete,
  InputAdornment,
  IconButton,
  FormHelperText,
  OutlinedInput,
  Link,
  CircularProgress,
} from '@mui/material';
import { LocationSearching } from '@mui/icons-material';
import { useFormik } from 'formik';
import { useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import * as yup from 'yup';

import { getAddress } from '../api/address';
import { geolocationSelector } from '../redux/ducks/geolocation';

const validationSchema = yup.object().shape({
  fullName: yup.string().required('Full Name is required'),
  email: yup
    .string()
    .email('Email must be a valid email')
    .required('Email Address is required'),
  streetAddress: yup.string().required('Street Address is required'),
  apt: yup.string(),
  city: yup.string().required('City is required'),
  country: yup.string().required('Country is required'),
  zip: yup.string().required('ZIP is required'),
});

export const Billing = () => {
  const navigate = useNavigate();
  const [isAddressLoading, setIsAddressLoading] = useState(false);

  const formik = useFormik({
    initialValues: {
      fullName: '',
      email: '',
      streetAddress: '',
      apt: '',
      city: '',
      country: '',
      zip: '',
    },
    validationSchema,
    onSubmit: (_) => {
      navigate('/payment');
    },
  });

  const countries = ['United States', 'Canada', 'Mexico', 'Ukraine'];

  const coords = useSelector(geolocationSelector);

  const handleGetAddress = async () => {
    if (coords && coords.latitude && coords.longitude) {
      try {
        setIsAddressLoading(true);
        const data = await getAddress(coords.latitude, coords.longitude);

        formik.setFieldValue('streetAddress', data.address.road || '');
        formik.setFieldValue('apt', data.address.house_number || '');
        formik.setFieldValue(
          'city',
          data.address.city || data.address.town || '',
        );
        formik.setFieldValue('country', data.address.country || '');
        formik.setFieldValue('zip', data.address.postcode || '');
      } catch (error) {
        console.error(error);
      } finally {
        setIsAddressLoading(false);
      }
    }
  };

  return (
    <form onSubmit={formik.handleSubmit}>
      <Stack
        direction="row"
        justifyContent="space-between"
        alignItems="flex-end"
        marginTop={3}
        marginBottom={2}
      >
        <Typography variant="h5" color="primary.main">
          Billing Information
        </Typography>
        <Link whiteSpace="nowrap" fontSize={12}>
          Same as shipping
        </Link>
      </Stack>
      <Stack spacing={2} marginBottom={2}>
        <Stack spacing={1}>
          <FormControl fullWidth>
            <FormLabel sx={{ marginBottom: 1, color: 'primary.main' }}>
              Billing Contact
            </FormLabel>
            <TextField
              name="fullName"
              placeholder="Full Name"
              size="small"
              fullWidth
              value={formik.values.fullName}
              onChange={formik.handleChange}
              onBlur={formik.handleBlur}
              error={formik.touched.fullName && Boolean(formik.errors.fullName)}
              helperText={formik.touched.fullName && formik.errors.fullName}
            />
          </FormControl>
          <FormControl fullWidth>
            <TextField
              name="email"
              placeholder="Email Address"
              size="small"
              fullWidth
              value={formik.values.email}
              onChange={formik.handleChange}
              onBlur={formik.handleBlur}
              error={formik.touched.email && Boolean(formik.errors.email)}
              helperText={formik.touched.email && formik.errors.email}
            />
          </FormControl>
        </Stack>
        <Stack spacing={1}>
          <FormControl fullWidth>
            <FormLabel sx={{ marginBottom: 1, color: 'primary.main' }}>
              Billing Address
            </FormLabel>
            <TextField
              name="streetAddress"
              placeholder="Street Address"
              size="small"
              fullWidth
              value={formik.values.streetAddress}
              onChange={formik.handleChange}
              onBlur={formik.handleBlur}
              error={
                formik.touched.streetAddress &&
                Boolean(formik.errors.streetAddress)
              }
              helperText={
                formik.touched.streetAddress && formik.errors.streetAddress
              }
            />
          </FormControl>
          <FormControl fullWidth>
            <TextField
              name="apt"
              placeholder="Apt, Suite, Bldg, Gate Code. (optional)"
              size="small"
              fullWidth
              value={formik.values.apt}
              onChange={formik.handleChange}
              onBlur={formik.handleBlur}
              error={formik.touched.apt && Boolean(formik.errors.apt)}
              helperText={formik.touched.apt && formik.errors.apt}
            />
          </FormControl>
        </Stack>
        <FormControl fullWidth>
          <OutlinedInput
            name="city"
            size="small"
            endAdornment={
              <InputAdornment position="end">
                <IconButton
                  aria-label="get location"
                  edge="end"
                  onClick={handleGetAddress}
                  disabled={isAddressLoading || !coords}
                >
                  {isAddressLoading ? (
                    <CircularProgress size={24} />
                  ) : (
                    <LocationSearching />
                  )}
                </IconButton>
              </InputAdornment>
            }
            placeholder="City"
            value={formik.values.city}
            onChange={formik.handleChange}
            onBlur={formik.handleBlur}
            error={formik.touched.city && Boolean(formik.errors.city)}
          />
          {formik.touched.city && formik.errors.city && (
            <FormHelperText sx={{ color: 'error.main' }}>
              {formik.errors.city}
            </FormHelperText>
          )}
        </FormControl>
        <Stack spacing={2} direction="row">
          <FormControl fullWidth>
            <Autocomplete
              freeSolo
              disablePortal
              options={countries}
              value={formik.values.country}
              getOptionLabel={(option) => option}
              onChange={(_, value) => formik.setFieldValue('country', value)}
              renderInput={(params) => (
                <TextField
                  {...params}
                  name="country"
                  size="small"
                  placeholder="Country"
                  onBlur={formik.handleBlur}
                  error={
                    formik.touched.country && Boolean(formik.errors.country)
                  }
                  helperText={formik.touched.country && formik.errors.country}
                />
              )}
            />
          </FormControl>
          <FormControl fullWidth>
            <TextField
              name="zip"
              placeholder="ZIP"
              size="small"
              fullWidth
              value={formik.values.zip}
              onChange={formik.handleChange}
              onBlur={formik.handleBlur}
              error={formik.touched.zip && Boolean(formik.errors.zip)}
              helperText={formik.touched.zip && formik.errors.zip}
            />
          </FormControl>
        </Stack>
      </Stack>
      <Button type="submit" variant="contained" fullWidth={false}>
        Continue
      </Button>
    </form>
  );
};
