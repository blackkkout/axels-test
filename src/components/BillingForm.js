import {
  Button,
  FormControl,
  FormLabel,
  Stack,
  Typography,
  TextField,
  Autocomplete,
  InputAdornment,
  IconButton,
  OutlinedInput,
  Link,
} from '@mui/material';
import { LocationSearching } from '@mui/icons-material';

export const BillingForm = ({ onSubmit }) => {
  const countries = ['United States', 'Canada', 'Mexico', 'Ukraine'];

  const handleSubmit = (event) => {
    event.preventDefault();
    onSubmit();
  };

  return (
    <form onSubmit={handleSubmit}>
      <Stack
        direction="row"
        justifyContent="space-between"
        alignItems="flex-end"
        marginTop={3}
        marginBottom={2}
      >
        <Typography variant="h5" color="primary.main">
          Billing Information
        </Typography>
        <Link whiteSpace="nowrap" fontSize={12}>
          Same as shipping
        </Link>
      </Stack>
      <Stack spacing={2} marginBottom={2}>
        <Stack spacing={1}>
          <FormControl fullWidth>
            <FormLabel sx={{ marginBottom: 1, color: 'primary.main' }}>
              Billing Contact
            </FormLabel>
            <TextField placeholder="Full Name" size="small" fullWidth />
          </FormControl>
          <FormControl fullWidth>
            <TextField placeholder="Email Address" size="small" fullWidth />
          </FormControl>
        </Stack>
        <Stack spacing={1}>
          <FormControl fullWidth>
            <FormLabel sx={{ marginBottom: 1, color: 'primary.main' }}>
              Billing Address
            </FormLabel>
            <TextField placeholder="Street Address" size="small" fullWidth />
          </FormControl>
          <FormControl fullWidth>
            <TextField
              placeholder="Apt, Suite, Bldg, Gate Code. (optional)"
              size="small"
              fullWidth
            />
          </FormControl>
        </Stack>
        <FormControl fullWidth>
          <OutlinedInput
            size="small"
            endAdornment={
              <InputAdornment position="end">
                <IconButton aria-label="get location" edge="end">
                  <LocationSearching />
                </IconButton>
              </InputAdornment>
            }
            placeholder="City"
          />
        </FormControl>
        <Stack spacing={2} direction="row">
          <FormControl fullWidth>
            <Autocomplete
              disablePortal
              options={countries}
              renderInput={(params) => (
                <TextField {...params} size="small" placeholder="Country" />
              )}
            />
          </FormControl>
          <FormControl fullWidth>
            <TextField placeholder="ZIP" size="small" fullWidth />
          </FormControl>
        </Stack>
      </Stack>
      <Button type="submit" variant="contained" fullWidth={false}>
        Continue
      </Button>
    </form>
  );
};
