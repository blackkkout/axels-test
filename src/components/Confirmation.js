import { Link, Stack, Typography } from '@mui/material';

export const Confirmation = () => (
  <Stack direction="column" gap={3}>
    <Typography variant="h5" color="primary.main">
      Thank you for your order!
    </Typography>
    <Stack direction="column" gap={1}>
      <Typography fontWeight={600}>Order number is: 123456789</Typography>
      <Typography lineHeight={1.25}>
        You will receive an email confirmation shortly to{' '}
        <Link>jonathan.smith@gmail.com</Link>
      </Typography>
    </Stack>
    <Typography lineHeight={1.25}>
      Estimated delivery day is <br />
      <b>Friday, July 15, 2022</b>
    </Typography>
    <Link fontSize={13}>Print Recipe</Link>
  </Stack>
);
