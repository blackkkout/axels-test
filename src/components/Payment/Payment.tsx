import { useNavigate } from 'react-router-dom';
import { useState } from 'react';
import {
  Button,
  FormControl,
  FormLabel,
  Stack,
  Typography,
  TextField,
  Grid,
  Alert,
} from '@mui/material';
import ShieldTwoToneIcon from '@mui/icons-material/ShieldTwoTone';
import { useFormik } from 'formik';
import * as yup from 'yup';

import { postPayment } from '../../api/orders';

const validationSchema = yup.object().shape({
  name: yup.string().required('Cardholder Name is required'),
  number: yup.string().required('Card Number is required'),
  exp: yup.string().required('Expiry Date is required'),
  cvc: yup.string().required('Security Code is required'),
});

export type PaymentFormValues = yup.InferType<typeof validationSchema>;

export const Payment = () => {
  const [error, setError] = useState<null | string>(null);
  const navigate = useNavigate();
  const formik = useFormik({
    initialValues: {
      name: '',
      number: '',
      exp: '',
      cvc: '',
    },
    validationSchema,
    onSubmit: async (values) => {
      try {
        const data = await postPayment(values);
        if (data?.success) {
          navigate('/confirmation');
        }
      } catch (error) {
        if (error instanceof Error) {
          setError(error.message);
        }
      }
    },
  });

  return (
    <form onSubmit={formik.handleSubmit}>
      <Typography
        variant="h5"
        color="primary.main"
        marginTop={3}
        marginBottom={2}
      >
        Payment
      </Typography>
      <Stack spacing={2} marginBottom={2}>
        <Stack direction="row" alignItems="center">
          <ShieldTwoToneIcon color="success" sx={{ marginRight: 1 }} />
          <Typography color="grey.500" fontSize={14}>
            This is a secure 128-bit SSL encrypted payment
          </Typography>
        </Stack>
        {error && <Alert severity="error">{error}</Alert>}
        <Stack spacing={1}>
          <FormControl fullWidth>
            <FormLabel sx={{ marginBottom: 1, color: 'primary.main' }}>
              Cardholder Name
            </FormLabel>
            <TextField
              name="name"
              placeholder="Name as it appears on your card"
              size="small"
              fullWidth
              value={formik.values.name}
              onChange={formik.handleChange}
              onBlur={formik.handleBlur}
              error={formik.touched.name && Boolean(formik.errors.name)}
              helperText={formik.touched.name && formik.errors.name}
            />
          </FormControl>
        </Stack>
        <Stack spacing={1}>
          <FormControl fullWidth>
            <FormLabel sx={{ marginBottom: 1, color: 'primary.main' }}>
              Card Number
            </FormLabel>
            <TextField
              name="number"
              placeholder="XXXX XXXX XXXX XXXX"
              size="small"
              fullWidth
              value={formik.values.number}
              onChange={formik.handleChange}
              onBlur={formik.handleBlur}
              error={formik.touched.number && Boolean(formik.errors.number)}
              helperText={formik.touched.number && formik.errors.number}
            />
          </FormControl>
        </Stack>
        <Grid container gap={4}>
          <Grid item xs={4}>
            <FormControl fullWidth>
              <FormLabel sx={{ marginBottom: 1, color: 'primary.main' }}>
                Expiry Date
              </FormLabel>
              <TextField
                name="exp"
                placeholder="MM / YY"
                size="small"
                fullWidth
                value={formik.values.exp}
                onChange={formik.handleChange}
                onBlur={formik.handleBlur}
                error={formik.touched.exp && Boolean(formik.errors.exp)}
                helperText={formik.touched.exp && formik.errors.exp}
              />
            </FormControl>
          </Grid>
          <Grid item xs={4}>
            <FormControl fullWidth>
              <FormLabel sx={{ marginBottom: 1, color: 'primary.main' }}>
                Security Code
              </FormLabel>
              <TextField
                name="cvc"
                size="small"
                fullWidth
                value={formik.values.cvc}
                onChange={formik.handleChange}
                onBlur={formik.handleBlur}
                error={formik.touched.cvc && Boolean(formik.errors.cvc)}
                helperText={formik.touched.cvc && formik.errors.cvc}
              />
            </FormControl>
          </Grid>
        </Grid>
      </Stack>
      <Button type="submit" variant="contained" fullWidth={false}>
        Pay Securely
      </Button>
    </form>
  );
};
