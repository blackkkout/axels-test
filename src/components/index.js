export { App } from './App';
export { BillingForm } from './BillingForm';
export { Order } from './Order';
export { OrderConfirmation } from './OrderConfirmation';
export { OrderItem } from './OrderItem';
export { OrderSummary } from './OrderSummary';
export { PaymentForm } from './PaymentForm';
export { ShippingForm } from './ShippingForm';
