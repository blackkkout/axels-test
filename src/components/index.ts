export { Billing } from './Billing';
export { Order } from './Order';
export { Confirmation } from './Confirmation';
export { Item } from './Item';
export { Summary } from './Summary';
export { Payment } from './Payment';
export { Shipping } from './Shipping';
