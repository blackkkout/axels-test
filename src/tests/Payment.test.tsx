import { render } from '@testing-library/react';

import { Payment } from '../components/Payment';

afterEach(() => {
  jest.resetAllMocks();
});

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => jest.fn(),
}));

describe('Payment', () => {
  it('should take a snapshot', () => {
    const view = render(<Payment />);
    expect(view).toMatchSnapshot();
  });
});
