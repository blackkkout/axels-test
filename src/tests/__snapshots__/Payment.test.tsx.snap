// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment should take a snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <h5
          class="MuiTypography-root MuiTypography-h5 css-1ue7way-MuiTypography-root"
        >
          Payment
        </h5>
        <div
          class="MuiStack-root css-1saxivv-MuiStack-root"
        >
          <div
            class="MuiStack-root css-1d9cypr-MuiStack-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorSuccess MuiSvgIcon-fontSizeMedium css-119npsf-MuiSvgIcon-root"
              data-testid="ShieldTwoToneIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 6.39v4.7c0 4 2.55 7.7 6 8.83 3.45-1.13 6-4.82 6-8.83v-4.7l-6-2.25z"
                opacity=".3"
              />
              <path
                d="M12 2 4 5v6.09c0 5.05 3.41 9.76 8 10.91 4.59-1.15 8-5.86 8-10.91V5zm6 9.09c0 4-2.55 7.7-6 8.83-3.45-1.13-6-4.82-6-8.83v-4.7l6-2.25 6 2.25z"
              />
            </svg>
            <p
              class="MuiTypography-root MuiTypography-body1 css-nh2nno-MuiTypography-root"
            >
              This is a secure 128-bit SSL encrypted payment
            </p>
          </div>
          <div
            class="MuiStack-root css-jfdv4h-MuiStack-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
            >
              <label
                class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1t6ae07-MuiFormLabel-root"
              >
                Cardholder Name
              </label>
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                    id=":r0:"
                    name="name"
                    placeholder="Name as it appears on your card"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-ihdtdm"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiStack-root css-jfdv4h-MuiStack-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
            >
              <label
                class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1t6ae07-MuiFormLabel-root"
              >
                Card Number
              </label>
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                    id=":r1:"
                    name="number"
                    placeholder="XXXX XXXX XXXX XXXX"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-ihdtdm"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container css-64wny6-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
              >
                <label
                  class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1t6ae07-MuiFormLabel-root"
                >
                  Expiry Date
                </label>
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                >
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                      id=":r2:"
                      name="exp"
                      placeholder="MM / YY"
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-ihdtdm"
                      >
                        <span
                          class="notranslate"
                        >
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
              >
                <label
                  class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1t6ae07-MuiFormLabel-root"
                >
                  Security Code
                </label>
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                >
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                      id=":r3:"
                      name="cvc"
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-ihdtdm"
                      >
                        <span
                          class="notranslate"
                        >
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-sghohy-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          Pay Securely
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <h5
        class="MuiTypography-root MuiTypography-h5 css-1ue7way-MuiTypography-root"
      >
        Payment
      </h5>
      <div
        class="MuiStack-root css-1saxivv-MuiStack-root"
      >
        <div
          class="MuiStack-root css-1d9cypr-MuiStack-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorSuccess MuiSvgIcon-fontSizeMedium css-119npsf-MuiSvgIcon-root"
            data-testid="ShieldTwoToneIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 6.39v4.7c0 4 2.55 7.7 6 8.83 3.45-1.13 6-4.82 6-8.83v-4.7l-6-2.25z"
              opacity=".3"
            />
            <path
              d="M12 2 4 5v6.09c0 5.05 3.41 9.76 8 10.91 4.59-1.15 8-5.86 8-10.91V5zm6 9.09c0 4-2.55 7.7-6 8.83-3.45-1.13-6-4.82-6-8.83v-4.7l6-2.25 6 2.25z"
            />
          </svg>
          <p
            class="MuiTypography-root MuiTypography-body1 css-nh2nno-MuiTypography-root"
          >
            This is a secure 128-bit SSL encrypted payment
          </p>
        </div>
        <div
          class="MuiStack-root css-jfdv4h-MuiStack-root"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1t6ae07-MuiFormLabel-root"
            >
              Cardholder Name
            </label>
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r0:"
                  name="name"
                  placeholder="Name as it appears on your card"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiStack-root css-jfdv4h-MuiStack-root"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1t6ae07-MuiFormLabel-root"
            >
              Card Number
            </label>
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r1:"
                  name="number"
                  placeholder="XXXX XXXX XXXX XXXX"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container css-64wny6-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
            >
              <label
                class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1t6ae07-MuiFormLabel-root"
              >
                Expiry Date
              </label>
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                    id=":r2:"
                    name="exp"
                    placeholder="MM / YY"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-ihdtdm"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
            >
              <label
                class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1t6ae07-MuiFormLabel-root"
              >
                Security Code
              </label>
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                    id=":r3:"
                    name="cvc"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-ihdtdm"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-sghohy-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="submit"
      >
        Pay Securely
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
